### ISSUE TITLE
CPU usage

### ISSUE THREAD

**Issue by azuremogul** [illiteratealliterator/manymine#7]
Thanks for this great container, which I have working with two 1.18 worlds, running on MacOS. 

A quick question though, please. The container seems to use quite a lot of CPU resources, even if no players are connected to the worlds. I typically see it fluctuate from every second between 0-50% CPU. Is this expected? I cannot see anything of relevance in the docker logs.

**Comment by illiteratealliterator** [comment]
Hi @azuremogul,

It's been a while since I've looked at the code for this. It was a bit of a weekend hack, but I've been meaning to revisit it since people actually seem to be using it. Looking at the code now, it does two things:

* Ping the known Bedrock servers once per second.
* Call the Docker API to find new servers once per second.

The ping should use basically no CPU, but I imagine the Docker API call could be doing something more intensive (perhaps more so on MacOS than on a Linux machine). I think calling that once per second is probably a little overzealous on my part.

When I get some time I plan on cleaning this up and I'll make those values configurable.

**Comment by azuremogul** [comment]
Thanks for looking into it. If I can help with testing, do let me know.

**Comment by illiteratealliterator** [comment]
Yeah it's a bit crap... not sure why I thought 1000ms was a good interval for updating the servers. It's a quick fix to change the interval, or to make it configurable, but I think the best solution is probably to just remove this feature and make the servers manually configured. I'll try and find some time soon to look at this, sorry!

**Comment by mmodler** [comment]
Or scan once at startup? If i create new servers, its no pain to restart manymine once?

**Comment by illiteratealliterator** [comment]
I've taken a look at this tonight and I've got some good news and some bad news.

The good news is that I found the CPU problem. It wasn't actually the Docker discovery process, but was instead the code that handles client pings unnecessarily recreating the Raknet serializer/deserializer objects. That's fixed now, so you should see significantly less CPU used once the container is up and running.

I made the discovery interval configurable via the env var MM_DISCOVERY_INTERVAL (in milliseconds), and if that's set to zero then it will only run discovery once on startup (which is now the default). The server ping interval (how often manymine pings your bedrock servers) is configurable via MM_PING_INTERVAL (minimum of 100ms) and defaults to 1000ms. I don't think there's much reason to change this one.

I've also updated to latest dependencies etc, and done some general clean up.

The bad news is hopefully just a "me problem". I don't know why, but manymine no longer works on my network, even using the existing image. For some reason the ping packets sent from a client (in my case an iPad running Minecraft) don't make it through to my manymine container. I can see the packets are coming into my Ubuntu VM where I run docker (using tcpdump), but for some reason they aren't making it through to the manymine app. I can manually send packets directly to it, and it will get packets sent from the iPad if I configure the server manually, so I believe it's something to do with broadcast packets not being routed to my container. It used to work, and I have no idea what's changed, so I'm a bit stumped.

Long story short, it'd be great if you could test my changes and see if it works for you. The new image is tagged 'test' (illiteratealliterator/manymine:test). Let me know how you go :)

**Comment by mmodler** [comment]
@illiteratealliterator I just tried 'test' without any configuration of the new params. manymine runs about 20 hours on my synology nas with 3 minecraft servers and i see 0,08% CPU usage now.

Great work, you're a genius :)

**Comment by illiteratealliterator** [comment]
Thanks @mmodler, really appreciate the testing, and the kind words - but I'm definitely no genius!

I'll push this new version to latest when I get a chance.

**Comment by azuremogul** [comment]
Just loaded up the test image and it is working well, thank you, with an average 0.3-0.5% CPU usage.

**Comment by illiteratealliterator** [comment]
@azuremogul - thanks for the testing, much appreciated!

I've pushed the new image to the `latest` tag and updated the readme. Thanks for reporting the issue!
